iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/KubernetesVerifier.java

```diff
@@ -72,38 +76,52 @@ public class KubernetesVerifier {
 
   private static final Logger LOG = LoggerFactory.getLogger(KubernetesVerifier.class);
   public static final Path BASE_DIR = Paths.get("src", "test", "resources", "checks");
-  private static final YamlParser YAML_PARSER = new YamlParser();
-
-  public static void verify(String templateFileName, IacCheck check) {
-    if (containsHelmContent(templateFileName)) {
-      HelmVerifier.verify(templateFileName, check);
-    } else {
-      Verifier.verify(YAML_PARSER, BASE_DIR.resolve(templateFileName), check,
-        multiFileVerifier -> new KubernetesTestContext(multiFileVerifier, new HelmInputFileContext(HelmVerifier.sensorContext,
-          inputFile(templateFileName, BASE_DIR))));
-    }
+  private static final SensorContextTester SENSOR_CONTEXT = SensorContextTester.create(BASE_DIR.toAbsolutePath());
+  private static final KubernetesAnalyzer KUBERNETES_ANALYZER = initializeKubernetesAnalyzer();
+  private static final TreeParser<Tree> PARSER = KUBERNETES_ANALYZER::parse;
+
+  public static void verify(String templateFileName, IacCheck check, String... fileNames) {
+    var initialization = initializeVerification(templateFileName, fileNames);
+    var inputFileContext = initialization.first();
+    var commentsVisitor = initialization.second();
+    var projectContext = prepareProjectContext(inputFileContext, fileNames);
+    Verifier.verify(PARSER, inputFileContext, check, multiFileVerifier -> new KubernetesTestContext(multiFileVerifier, inputFileContext, projectContext), commentsVisitor);
   }
 
-  public static void verify(String templateFileName, IacCheck check, Verifier.Issue... expectedIssues) {
-    if (containsHelmContent(templateFileName)) {
-      HelmVerifier.verify(templateFileName, check, expectedIssues);
-    } else {
-      Verifier.verify(YAML_PARSER, BASE_DIR.resolve(templateFileName), check, expectedIssues);
-    }
+  public static void verify(String templateFileName, IacCheck check, Collection<Verifier.Issue> expectedIssues) {
+    var initialization = initializeVerification(templateFileName);
+    var inputFileContext = initialization.first();
+    var commentsVisitor = initialization.second();
+    var projectContext = prepareProjectContext(inputFileContext);
+    Verifier.verify(PARSER, inputFileContext, check, multiFileVerifier -> new KubernetesTestContext(multiFileVerifier, inputFileContext, projectContext), commentsVisitor,
```

Comment:
`multiFileVerifier -> new KubernetesTestContext(multiFileVerifier, inputFileContext, projectContext)`
Maybe adding this in its own line will be better